### PR TITLE
PL.Editor #59 Test for "does not alter new line spaced lists"

### LINKED ISSUE #59: Toggling between Rich and Markdown resets all numbers in a numbered list to `1.`
Really enjoying the new Editor, just wanted to point out that when toggling between the rich mode and markdown (which i did to try to get more control on how some code bits were displaying), resulted in some loss of formatting when back in rich mode. 

The automated indentations were still present, but the numbers all reset to 1. It was easy to fix this by placing my cursor after the automatic number at the start of my text, and click backspace and the numbering caught up to where it should have been: 

screen shot 2016-10-26 at 12 11 21 pm

screen shot 2016-10-26 at 12 11 32 pm

Also lost when returning to rich mode: the carriage returns (which i placed after each numbered item to space out the bottom of the screenshot from the top of the next instruction) were deleted

### PR DESCRIPTION
**Pull request by rexagod**
Fixes #59 

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] PR body includes `fixes #0000`-style reference to original issue #
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. 

Thanks!

### CODE REVIEW

**Comment by jywarren** [comment]
Huh, interesting! Do you know why this didn't reproduce the issue I'd made a gif of in #59? Would it be different if we triggered it by simulating a button press?

**Comment by rexagod** [comment]
@jywarren I implemented this for the list given below.

> 
> I think a number of things could be done to progress this - perhaps a test which enters a list like:
> ```
> 1. one
> 
> 2. two
> 
> 3. three
> ```
> and looks to see that these are undisturbed after being transformed back and forth from wysiwyg mode to markdown mode and back.

Do you want the test to cover this instead?
```
 1. one
 
 2. two
 
 3. three

 4.
 5.
 6.
 ```

**Comment by jywarren** [comment]
Thanks! I think i'd like the test to reproduce as much as possible the error we documented in the gif; my proposal for the list you quoted was an attempt at that... but maybe the correct test string would be:

```
1. one
2. two
3. three

 1.

 2.

 3.

```
What I think happens is that either `domador` or `megamark` eats the extra carriage returns, turning the final 3 lines into part of the same list as above. Then because there is no content in them, they get mushed into the same line. It's pretty bad! But let's see if we can reproduce it.

File: spec/javascripts/rich_text_module_spec.js [hunk 1]
```diff
@@ -134,5 +147,26 @@ describe("RichTextModule", function() {
...
+
+    module.value('<p>1. first<br/>2. second<br/>3. third</p><p>1.</p><p>2.</p><p>3.</p>')
+
+    module.setMode('markdown')
+
+    expect(module.html()).toContain('<p>1. first<br>2. second<br>3. third</p><p>1.</p><p>2.</p><p>3.</p>')
+    
+    module.setMode('markdown')
+    
+    module.value('1. one\n2. two\n3. three\n\n1.\n\n2.\n\n3.')
```
Comment (jywarren) [inline spec/javascripts/rich_text_module_spec.js]: Take a look here - i entered this as in the string, convert to wysiwyg and back to markdown, and hope to find the same string there. But I kind of expect I won't... let's see!

**Comment by rexagod** [comment]
Okay so this gives the following error.

````
X does not alter new line spaced lists
     Expected '1. one
2. two
3. three
4. 5. 6.' to be '1. one
2. two
3. three

1.

2.

3.'
````
@jywarren Did you mean this?
````js
expect(module.value()).not.toBe('1. one\n2. two\n3. three\n\n1.\n\n2.\n\n3.')
````
Correct me if I'm wrong, but I think `\n` won't work in this case since `domador` and `woofmark` fail to recogize it, `woofmark` interprets this as a single list and `domador` adds escape sequences to `\n`.
***
test

**Pull request closed without merging**